### DISCUSSION TITLE
New Aqara Motion Sensor P1 (MS-S02) attribute help

### DISCUSSION THREAD

**Discussion by rebel7580** [arendst/Tasmota#18201] [Q&A]
I just got an "upgraded" Aqara "Motion Sensor P1", labled "MS-S02" on the sensor. 

With my "old" sonoff bridge running 12.1.1.1, it paired OK.

ZbInfo:
```
{"Device":"0xA378","Name":"MotionUtil","IEEEAddr":"0x54EF441000773B0C","ModelId":"lumi.motion.ac02","Manufacturer":"LUMI","Endpoints":[1],"Config":[],"Reachable":true,"BatteryPercentage":100,"BatteryLastSeenEpoch":1678832341,"LastSeen":34,"LastSeenEpoch":1678912912,"LinkQuality":68}
```

When detecting motion, it reports:
```
{"Device":"0xA378","Name":"MotionUtil","FCC0/0112":65628,"Endpoint":1,"LinkQuality":76}
```

I think I've figured out that the value reported for FCC0/0112 is 5 hex digits of the format OLLLL, where "O" is Occupancy (1=detected), and LLLL is Illuminance of some arbitrary units.

I'd like to use a plugin to change this, at least the Occupancy, preferably both, but can't figure out how to do this in a plugin. 

Essentially, for Occupancy, "shift right 16 bits" so div:65535?
```
FCC0/0112,Occupancy,div:65535
```
for Illuminance, "mask with 0xFFFF", could be:
```
FCC0/0112,Illuminance,%21
```
If each of these is correct, can I define both operations on the same attribute to generate two results? I tried this:
```
:lumi.motion.ac02*,
FCC0/0112=0500/FFF3,div:65535
FCC0/0112=0500/FFF4,%21
0500/FFF3,Occupancy
0500/FFF4,Illuminance
```
but get an invalid header error.

BTW, I don't notice the sensor ever reporting a detection timeout, although it is supposed to have a 1-200 second timeout (settable via command although I don't know what that command might be). Also, there is a command to set sensitivity (high medium low).
I don't know what the out-of-box defaults are for either.

**Comment by rebel7580** [comment]
Zigbee2MQTT has support details here:
https://www.zigbee2mqtt.io/devices/RTCGQ14LM.html

**Comment by rebel7580** [comment]
I found my error: left out the initial header (#Z2Tv1); The error is gone, did zbload of the file and zbloaddump shows it properly. But still don't see a change. Here's my latest attempt:
```
#Z2Tv1
#AqaraMotion Sensor P1: MS-S02
:lumi.motion.ac02*,
FCC0/0112=0500/FFF3,div:65535
0500/FFF3,Occupancy
```
I took out the illuminance part, but this one still only shows "FCC0/0112":65618, for example.

**Comment by rebel7580** [comment]
Upgraded to 12.4.0.2, and Occupancy now shows up!

Tried to get illuminance by adding this:
```
FCC0/0112=0500/FFF4,%21
0500/FFF4,Illuminance
```
thinking it would just use the last 16 bits (to remove the occupancy bit), but no luck so far. Ideally, I'd need to subtract 65536, but that doesn't seem possible with the add property.

**Comment by rebel7580** [comment]
@s-hadinger 
Hi -

I know you are busy with more interesting newer things like Matter and Berry, but if you get a chance, could you look at support for this new version of the Aqara motion sensor?

According to discussion on Zigbee2MQTT, they have full support for the "standard" attributes for a motion sensor - occupancy. illumination, battery, etc. plus writing timeout and sensitivity values to it.

I got Occupancy to work via a plugin, but nothing else. Oddly, I don't see any Occupancy=0 messages.
A message with Occupancy=1 is all I need for now. I timeout via my external controller anyway. So I have it working.

But it would be nice to get the rest of the attributres reported, and even be able to set the timeout and senitivity, which are new for this version. I took a look at what Zigbee2MQTT was doing, but it's beyond me in terms of how to apply that to Tasmota! However, I'd be happy to help where I can - logs, testing, etc.

Thanks!

**Reply by s-hadinger** [reply]
Is your gateway ESP32 based? If so we can try to add some Berry logic for this specific case.

**Reply by rebel7580** [reply]
To summarize what I've done so far:

Original report when detecting motion:
```
{"Device":"0xA378","Name":"MotionUtil","FCC0/0112":65628,"Endpoint":1,"LinkQuality":76}
```

After I applied the following plugin:
```
#Z2Tv1
#AqaraMotion Sensor P1: MS-S02
:lumi.motion.ac02*,
FCC0/0112=0500/FFF3,div:65535
FCC0/0112=0500/FFF4,add:-32768
0500/FFF3,Occupancy
0500/FFF4,Illuminance
```
ZbLoadDump:
```
12:22:21.147 # imported from '<internal_plugin>'
12:22:21.148 :TRADFRI*,
12:22:21.148 :SYMFONISK*,
12:22:21.149 0001/0021=0001/0021,mul:2
12:22:21.150 # imported from '<internal_plugin>'
12:22:21.151 :lumi.sensor_magnet*,
12:22:21.152 0006/0000=0500/FFF2
12:22:21.153 # imported from '/aqaramss02.fb'
12:22:21.153 :lumi.motion.ac02*,
12:22:21.154 0500/FFF3,Occupancy
12:22:21.155 0500/FFF4,Illuminance
12:22:21.156 FCC0/0112=0500/FFF3,div:65535
12:22:21.157 FCC0/0112=0500/FFF4,add:-32768
12:22:21.158 <==== END
```
Resulting report when detecting motion (Note only Occupancy shows, no Illumination):

```
{"Device":"0xA378","Name":"MotionUtil","Occupancy":1,"Endpoint":1,"LinkQuality":63}
```

Periodic reports from device:
```
{"Device":"0xA378","Name":"MotionUtil","FCC0/00F7":"01211E0C0328110421000005210100082106010A2154230C200113200014200064100065212C0069201E6A20026B2000","Endpoint":1,"LinkQuality":76}
```
Note: no battery info.


FWIW, here's what I get from the "older" version of the Aqara motion sensor:
```
{"Device":"0xED02","Name":"Motion5","0000/FF01":"0121D10B03281C0421A84305210C00062404000000000A2169FE6410000B210100","BatteryVoltage":3.03,"BatteryPercentage":100,"Xiaomi_64":0,"Endpoint":1,"LinkQuality":102}
```

**Reply by rebel7580** [reply]
@s-hadinger   OK, I get that it's too much for full support in the 826x device, but is a plugin possible to extract both Occupancy and Illuminance from FCC0/0112? I tried that (as above) but suspect the plugin only maps FCC0/0112 once and then ignores any remaining attempts to map it to a second attribute.  Having both Occupancy and Illuminance would be better. 

In any case, thanks for your great work with Z2T!

**Comment by rebel7580** [comment]
No, unfortunately, it’s the older 8266 version.
